Patient Registry System/PatientRegistrySystem.Services/UserService.cs

```diff
@@ -0,0 +1,58 @@
+﻿using AutoMapper;
+using PatientRegistrySystem.DB.Entities;
+using PatientRegistrySystem.DB.Repos;
+using PatientRegistrySystem.Domain.Dto;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientRegistrySystem.Services
+{
+    public class UserService : IUserService
+    {
+        private readonly IMapper _mapper;
+        private readonly IUserRepository _userRepository;
+
+        public UserService(IMapper mapper, IUserRepository userRepository)
+        {
+            _mapper = mapper;
+            _userRepository = userRepository;
+        }
+
+        public async Task<IEnumerable<UserDto>> GetAllUsersAsync()
+        {
+            var userFromDB = await _userRepository.GetAllAsync();
+            if (!userFromDB.Any()) { return null; }
+            return _mapper.Map<UserDto[]>(userFromDB);
```

Comment:
we should do the mapping from DB to Dto in the DB repository. So GetAllAsync should return domain models not db models. And same applies to the GenericRepository. Services layer shouldn't see these DB model